### PR TITLE
specify testing-type experiments

### PR DESCRIPTION
**Pull request by emilyrohrbough**
per: https://cypressio.slack.com/archives/C03STSRDZU1/p1661270190081439

### CODE REVIEW

**Review by lmiller1990 — APPROVED** [review]
Looks good, thanks... one nit: we use "end-to-end", "End-to-end" and "End-to-End" in the docs. We should standardize this, my vote is "End to End", but another battle for another day.

**Comment by emilyrohrbough** [comment]
@lmiller1990 I copied & pasted from the config page, but good catch. The [Style Guide](https://docs.google.com/document/d/1pOZ0Ntm4b1Eev_iY5vLBPhOxx66Z8Njrvvg6WMCvWRI/edit#) has "Cypress End-to-End Testing" listed.

File: content/guides/references/experiments.md [hunk 1]
```diff
@@ -26,10 +26,31 @@ configuration to Cypress.
...
-| `experimentalSingleTabRunMode`                | `false` | Run all specs in a single tab, instead of creating a new tab per spec. This can improve run mode performance, but can impact spec isolation and reliability on large test suites. This experiment currently only applies to Component Testing.                                                                                 |
 | `experimentalSourceRewriting`                 | `false` | Enables AST-based JS/HTML rewriting. This may fix issues caused by the existing regex-based JS/HTML replacement algorithm. See [#5273](https://github.com/cypress-io/cypress/issues/5273) for details.                                                                                                                         |
 
+## Testing Type-Specific Experiments
+
+You can provide configuration options for either E2E or Component Testing by
+creating `e2e` and `component` objects inside your Cypress configuration.
+
+### e2e
```
Comment (emilyrohrbough) [inline content/guides/references/experiments.md:36]:
```suggestion
### End-to-End Testing
```

File: content/guides/references/experiments.md [hunk 2]
```diff
@@ -26,10 +26,31 @@ configuration to Cypress.
...
+
+These experiments are available to be specified inside the `e2e` configuration
+object:
+
+| Option                         | Default | Description                                                                                                                                                                                                  |
+| ------------------------------ | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `experimentalSessionAndOrigin` | `false` | Enables cross-origin and improved session support, including the [`cy.origin()`](/api/commands/origin) and [`cy.session()`](/api/commands/session) commands. This enables `testIsolation=strict` by default. |
+
+### component
```
Comment (emilyrohrbough) [inline content/guides/references/experiments.md:45]:
```suggestion
### Component Testing
```

**Review by chrisbreiding — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**